compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -530,6 +537,11 @@ pub fn compile_declarative_macro(
         None => {}
     }
 
+    // Compute the spans of the macro rules
+    // We only take the span of the lhs here,
+    // so that the spans of created warnings are smaller.
+    let rule_spans = lhses.iter().map(|lhs| lhs.span()).collect::<Vec<_>>();
```

Comment:
Only need to collect spans if `id != DUMMY_NODE_ID`, otherwise it's a macro decoded from some other crate.